src/data/mystery-encounter.ts

```diff
@@ -135,7 +145,90 @@ export default class MysteryEncounter implements MysteryEncounter {
    * @returns
    */
   meetsRequirements?(scene: BattleScene) {
-    return !this.requirements.some(requirement => !requirement.meetsRequirement(scene));
+    const sceneReq =  !this.requirements.some(requirement => !requirement.meetsRequirement(scene));
```

Comment:
if you inline these calls in the return statement, JS will stop executing checks early if one of them returns false. Could just separate the checks with newlines.

Reply:
i originally had a bunch of logging for each set of requirements. If we ever want to add logging again to debug event triggering, keeping it like this will be easier, no?

Reply:
imma resolve this without changing, lmk if you feel strongly